vm/systemSmartContracts/esdt.go

```diff
@@ -747,13 +761,13 @@ func (e *esdt) esdtControlChanges(args *vmcommon.ContractCallInput) vmcommon.Ret
 	return vmcommon.Ok
 }
 
-func (e *esdt) saveToken(token *ESDTData) error {
+func (e *esdt) saveToken(identifier []byte, token *ESDTData) error {
 	marshaledData, err := e.marshalizer.Marshal(token)
 	if err != nil {
 		return err
 	}
 
-	e.eei.SetStorage(token.TokenName, marshaledData)
+	e.eei.SetStorage(identifier, marshaledData)
```

Comment:
getExistingToken - method shoudla also be changed with argument identifier instead of tokenName. Also the if token not found the error should be ErrNoTokenWithGivenIdentifier instead of ErrNoTokenWithGivenName

Reply:
done.

Reply:
ok